src/inline_array/mod.rs

```diff
@@ -608,15 +627,68 @@ mod test {
         let mut chunk: InlineArray<String, [u8; 512]> = InlineArray::new();
         chunk.push("Hello".to_owned());
         assert_eq!(chunk[0], "Hello");
+
+        let mut chunk: InlineArray<String, [u16; 512]> = InlineArray::new();
+        chunk.push("Hello".to_owned());
+        assert_eq!(chunk[0], "Hello");
+    }
+
+    #[test]
+    fn unsufficient_alignment1() {
+        #[repr(align(256))]
+        struct BigAlign(u8);
+        #[repr(align(32))]
+        struct MediumAlign(u8);
+
+        assert_eq!(0, InlineArray::<BigAlign, [usize; 256]>::CAPACITY);
+        assert_eq!(0, InlineArray::<BigAlign, [u64; 256]>::CAPACITY);
+        assert_eq!(0, InlineArray::<BigAlign, [f64; 256]>::CAPACITY);
+        assert_eq!(0, InlineArray::<BigAlign, [MediumAlign; 256]>::CAPACITY);
     }
 
     #[test]
-    fn big_align_elem() {
+    #[should_panic(
+        expected = "InlineArray can't satisfy alignment of sized_chunks::inline_array::test::unsufficient_alignment2::BigAlign"
+    )]
+    fn unsufficient_alignment2() {
         #[repr(align(256))]
         struct BigAlign(usize);
 
         let mut chunk: InlineArray<BigAlign, [usize; 256]> = InlineArray::new();
         chunk.push(BigAlign(42));
```

Comment:
AFAIK the part that'll panic is the `InlineArray::new()`. But this tests suggests the intention is to panic in push?

Reply:
I wasn't sure when it should panic. It'd be possible to just set CAPACITY=0 when alignment is wrong, and this would make `new` succeed and fail only at `push`.